extensions/src/main/java/org/pentaho/platform/web/http/api/resources/ActionResource.java

```diff
@@ -88,14 +86,14 @@ public Response invokeAction(
     @QueryParam( ActionUtil.INVOKER_ACTIONID ) String actionId,
     @QueryParam( ActionUtil.INVOKER_ACTIONCLASS ) String actionClass,
     @QueryParam( ActionUtil.INVOKER_ACTIONUSER ) String actionUser,
-    final String actionParams ) {
+    final ActionParams actionParams ) {
```

Comment:
Also, I find String to me more opaque  then an explicit data model.  A string has no defined structure making it hard to know what correct value is. Since its a Rest endpoint that accepts JSON it makes sense to let the Jax layer convert json to the POJO.